src/services/adService.ts

```diff
@@ -0,0 +1,69 @@
+import { IAdModel } from '../db/';
+import { getByFilter } from '../repositories/adRepository';
+
+export const getAdsByFilter = async (
+  filter?: any,
+  limit?: number,
+  skip?: number,
+  sort?: string
+): Promise<IAdModel[]> => {
+  const searchFilter: any = {};
+  if (filter.bodyTypeId) {
+    searchFilter.bodyTypeId = { $in: [...filter.bodyTypeIds] };
```

Comment:
What is filter is undefined?

Reply:
Filter do not may be undefined as we pre validate request